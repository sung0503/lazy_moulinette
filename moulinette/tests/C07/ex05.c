#include <stdio.h>

char **ft_split(char *str, char *charset);

void test(char *str, char *charset)
{
    int i = -1;
    char **res;
    res = ft_split(str, charset);
    printf("---\n");
    while (res[++i] != 0)
    {
        printf("%s\n", res[i]);
    }
}

int main(void)
{
    test("", "");
    test("Hello 42 Gyeongsan So Hot Place.", "");
    test("Hello 42 Gyeongsan So Hot Place.", "!@#$");
    test("Hello 42 Gyeongsan So Hot Place.", ". ");
    test("Hello 42 Gyeongsan So Hot Place.", "4");
    test("Hello 42 Gyeongsan So Hot Place.", "e");
    test("aaaa", "a");
    test("aaabbbaaabbbaaa", "a");
    test("aaabbbaaabbbaaa", "ab");
    test("bsaasbsabsabsabasbsabsabsabsabsabsabssbsbaasbsabsabsabsabsabbbbbbasbs", "abs");
    test("bsaasbsabsabsabasbccccsabsabsabsabsabsabssbsbaasbsabsabsabsabsabbbbbb", "abs");
    test("bcahhhbcahhhbcah h$$#%^#^hhbca", "abc");
    return 0;
}
