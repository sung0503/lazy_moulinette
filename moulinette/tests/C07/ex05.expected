---
---
Hello 42 Gyeongsan So Hot Place.
---
Hello 42 Gyeongsan So Hot Place.
---
Hello
42
Gyeongsan
So
Hot
Place
---
Hello 
2 Gyeongsan So Hot Place.
---
H
llo 42 Gy
ongsan So Hot Plac
.
---
---
bbb
bbb
---
---
---
cccc
---
hhh
hhh
h h$$#%^#^hh
